motionplan/plan.go

```diff
@@ -307,3 +307,37 @@ func (e *ExecutionState) CurrentInputs() map[string][]referenceframe.Input {
 func (e *ExecutionState) CurrentPoses() map[string]*referenceframe.PoseInFrame {
 	return e.currentPose
 }
+
+// CalculateFrameErrorState takes an ExecutionState and a Frame and calculates the error between the Frame's expected
+// and actual positions.
+func CalculateFrameErrorState(e ExecutionState, frame referenceframe.Frame) (spatialmath.Pose, error) {
+	currentInputs, ok := e.CurrentInputs()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currentPose, ok := e.CurrentPoses()[frame.Name()]
+	if !ok {
+		return nil, fmt.Errorf("could not find frame %s in ExecutionState", frame.Name())
+	}
+	currPoseInArc, err := frame.Transform(currentInputs)
+	if err != nil {
+		return nil, err
+	}
+	path := e.Plan().Path()
+	if path == nil {
+		return nil, errors.New("cannot calculate error state on a nil Path")
+	}
+	index := max(e.Index()-1, 0)
+	if index < 0 || index >= len(path) {
+		return nil, fmt.Errorf("index %d out of bounds for Path of length %d", index, len(path))
+	}
+	pose, ok := path[index][frame.Name()]
```

Comment:
This isn't correct. The pose on the path is the ending pose of the step, not the starting pose of the step.

Reply:
I think this should be fine because I was taking the index minus one